### PR TITLE
Forecast endpoint

### PR DESCRIPTION
**Pull request by nancylee713**
This PR adds the following:

- Within GET request for the `/` endpoint, I made 3 async functions:
   - `fetchCoords()`  to get coordinates from google geocoding api
   - `fetchForecastData()` to get forecast data
   - `formatData()` to retrieve specific attributes using 3 kinds of pojos (forecast, hourly, and daily).

- To check whether the request contains a correct api key, I added a flow control statement to render either response data or error message.

As of now, `User.find(key)` returns a builder object, not the record that I'm looking for from the users table.

### CODE REVIEW

**Comment by nancylee713** [comment]
@evettetelyas

File: db/migrations/20191120190404_user.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+exports.up = function(knex) {
+  return Promise.all([
+    knex.schema.createTable('users', function (table) {
+      table.increments('id').primary();
+      table.string('email');
+      table.string('password');
+      table.string('apiKey');
+      table.string('favoriteLocation');
```
Comment (evettetelyas) [inline db/migrations/20191120190404_user.js]: users can have multiple favorite locations, might make things easier down the line to create a favorites table with a location and userId

File: lib/models/users.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+class User {
+  constructor(obj) {
+    this.email = obj.email,
+    this.password = obj.password,
+    this.apiKey = obj.apiKey,
+    this.favoriteLocation = obj.favoriteLocation;
+  }
+
+  static find(key) {
+    return database("users").where({ apiKey: key });
```
Comment (evettetelyas) [inline lib/models/users.js]: YES! i had to go backwards and do this. smart move!!!

File: routes/api/v1/forecast.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        let latLong = await geocodingData["results"][0]["geometry"]["location"];
+
+        return latLong;
+      } catch (err) {
+        console.log(err)
+      }
+    }
+
+    fetchForecastData = async () => {
+      const key = await yaml.safeLoad(fs.readFileSync("./config.yml", "utf8"))["dark_sky_api"];
```
Comment (evettetelyas) [inline routes/api/v1/forecast.js]: woah! this is COOL

File: db/migrations/20191120190404_user.js (hunk 1 above)
Comment (nancylee713) [reply db/migrations/20191120190404_user.js]: I see. Creating a separate table for the favorites seems like a much better design.

File: lib/models/users.js (hunk 2 above)
Comment (nancylee713) [reply lib/models/users.js]: Thanks! But I'm still not sure if this function returns a boolean (ie. whether this user exists)?

File: routes/api/v1/forecast.js (hunk 3 above)
Comment (nancylee713) [reply routes/api/v1/forecast.js]: Yes, the package is called `js-yaml`, similar to `application.yml` in a rails app

**Pull request merged**